tv_apps/smart-home/js/card_picker.js

```diff
@@ -248,6 +250,18 @@
       function appendToGridView(appCardElem) {
         that.gridView.appendChild(appCardElem);
         that.appCardElems.push(appCardElem);
+
+        if (top >= 0) {
```

Comment:
I would prefer to move this paragraph out of `appendToGridView`, and let it be an independent for-loop. After you found the first row you can just break the for-loop. I think that should be more easier to read.